includes/exceptions/class-amount-too-low-exception.php

```diff
@@ -0,0 +1,54 @@
+<?php
+/**
+ * Class API_Exception
+ *
+ * @package WooCommerce\Payments
+ */
+
+namespace WCPay\Exceptions;
+
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * Class representing API_Exception
+ */
+class Amount_Too_Low_Exception extends API_Exception {
+
+	/**
+	 * Constructor
+	 *
+	 * @param string    $message         The Exception message to throw.
+	 * @param string    $error_code      Error code returned by the server, for example wcpay_account_not_found.
+	 * @param int       $http_code       HTTP response code.
+	 * @param array     $additional_data Additional data for the exception.
+	 * @param int       $code            The Exception code.
+	 * @param Throwable $previous        The previous exception used for the exception chaining.
+	 */
+	public function __construct( $message, $error_code, $http_code, $additional_data = null, $code = 0, $previous = null ) {
+		$message = __(
+			'The specified amount is less than the minimum amount allowed. Use a higher amount and try again.',
+			'woocommerce-payments'
+		);
+		if ( is_array( $additional_data ) && isset( $additional_data['min_amount'], $additional_data['currency'] ) ) {
+			$min_amount = $additional_data['min_amount'];
+			$currency   = strtolower( $additional_data['currency'] );
+
+			if ( ! in_array( $currency, \WC_Payments_Utils::zero_decimal_currencies(), true ) ) {
+				$min_amount = $min_amount / 100;
+			}
```

Comment:
Re-used `interpret_stripe_amount`, but I am not sure if I understand the part about the `wc_price`.

Reply:
> Re-used interpret_stripe_amount
👍 

>  I am not sure if I understand the part about the wc_price.

Sorry, for the vague comment. I was thinking about creating utility method like `\WC_Payments_Utils::format_currency( $amount, $currency )` which would check if currency is zero decimal and adjust amount as needed. Then in this error handler we could use it right inside sprintf like this: `\WC_Payments_Utils::format_currency( $additional_data['min_amount'], additional_data['currency'] )`. 

This would allow us to move away from using `wc_price` directly everywhere in the code base and with time replace `wc_price` with own implementation at all (to solve <span> problem). 

What do you think?

Reply:
> Sorry, for the vague comment. I was thinking about creating utility method like \WC_Payments_Utils::format_currency( $amount, $currency ) 

I like this approach. It would also address our `<span>` discussion. As a first naive implementation I would just call `wc_price` wrapped with `strip_tags` to remove any HTML formatting.

Reply:
I am still on the fence with the currency symbol. `wc_price` will try to use an HTML-entity for a currency symbol and my current implementation of `format_amount` should convert it into a (Unicode) character (depends on a symbol). Which may play nicely with how it is later used in the notice but may be not.